src/components/annotations/convert_coords.js

```diff
@@ -26,6 +26,8 @@ var toLogRange = require('../../lib/to_log_range');
  *     same relayout call should override this conversion.
  */
 module.exports = function convertCoords(gd, ax, newType, doExtra) {
+    ax = ax || {};
```

Comment:
.. so that e.g. in [`relayout`](https://github.com/plotly/plotly.js/blob/9f4823be4f50239503bb3c5da126db3980619384/src/plot_api/plot_api.js#L1991-L1992) blank axis objects that pass the forgiving [`AX_NAME_PATTERN`](https://github.com/plotly/plotly.js/blob/9f4823be4f50239503bb3c5da126db3980619384/src/plots/cartesian/constants.js#L30) (which also matches `zaxis?` by the way) don't leave to exceptions.